weave/scorers/moderation_scorer.py

```diff
@@ -20,22 +55,271 @@ class OpenAIModerationScorer(LLMScorer):
+    category_threshold: int = 2
+    max_tokens: int = 512
+    overlap: int = 50
+    _categories: list[str] = PrivateAttr(
+        default=[
+            "Race/Origin",
+            "Gender/Sex",
+            "Religion",
+            "Ability",
+            "Violence",
+        ]
+    )
+
+    def load_model(self) -> None:
+        if self.base_url:
+            print(f"Using external API at {self.base_url} for scoring.")
+            return  # Skip local model loading if base_url is provided
+        try:
+            from transformers import AutoModelForSequenceClassification
+        except ImportError:
+            print(
+                "The `transformers` package is required to use {self.__class__.__name__}, please run `pip install transformers`"
+            )
+        """Initialize the toxicity model and tokenizer."""
+        if os.path.isdir(self.model_name_or_path):
+            self._local_model_path = self.model_name_or_path
+        else:
+            self._local_model_path = download_model(MODEL_PATHS["toxicity_scorer"])
+
+        self.model = AutoModelForSequenceClassification.from_pretrained(
+            self._local_model_path, device_map=self.device, trust_remote_code=True
+        )
+        self.model.eval()
+
+    def load_tokenizer(self) -> None:
+        if self.base_url:
+            print(f"Using external API at {self.base_url} for scoring.")
+            return  # Skip local model loading if base_url is provided
+        try:
+            from transformers import AutoTokenizer
+        except ImportError:
+            print(
+                "The `transformers` package is required to use {self.__class__.__name__}, please run `pip install transformers`"
+            )
+        self.tokenizer = AutoTokenizer.from_pretrained(self._local_model_path)
+        print(f"Model and tokenizer loaded on {self.device}")
+
+    def predict_chunk(self, input_ids: "Tensor") -> list[Union[int, float]]:
+        """
+        Predict toxicity scores for a chunk of tokenized input.
+
+        Args:
+            input_ids: Tokenized input IDs for the chunk.
+
+        Returns:
+            A list of prediction scores for each category.
+        """
+        import torch
+
+        with torch.inference_mode():
+            attention_mask = (input_ids != 0).long()
+            outputs = self.model(input_ids=input_ids, attention_mask=attention_mask)
+            predictions = outputs.logits.argmax(dim=-1).squeeze().tolist()
+        if isinstance(predictions, int):
+            return [predictions]
+        return predictions
+
+    def _score_via_api(self, output: str) -> dict[str, Any]:
+        import requests
+
+        assert self.base_url is not None
+        response = requests.post(self.base_url, json={"output": output})
+        response.raise_for_status()
+        return response.json()
+
+    @weave.op
+    def score(self, output: str) -> dict[str, Any]:
+        # remote scoring
+        if self.base_url:
```

Comment:
Something about this feels weird.  There are 2 cases here:
1. The model is running locally (no base URL)
2. The model is running remotely (with base URL)

They kinda feel like two different flavours of scorers... wondering if there's a better way to express this